File: go/chapter10/remember_me.go
package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
)

func main() {

	displayUserInfo()
	// saveUserInfo()
}

func saveUserInfo() {
	scanner := bufio.NewScanner(os.Stdin)
	user := struct {
		name    string
		age     int
		contact string
	}{}

	fmt.Printf("Enter name: ")
	scanner.Scan()
	user.name = scanner.Text()

	fmt.Printf("Enter age: ")
	scanner.Scan()
	intAge, err := strconv.ParseInt(scanner.Text(), 10, 64)
	if err != nil {
		fmt.Println("Please enter a valid age!!")
		return
	}
	user.age = int(intAge)

	fmt.Printf("Enter contact: ")
	scanner.Scan()
	user.contact = scanner.Text()

	userBytes, _ := json.Marshal(map[string]interface{}{"name": user.name, "age": user.age, "contact": user.contact})
	os.WriteFile("./remember.json", userBytes, os.ModePerm)
}

func displayUserInfo() {
	userFile, err := os.ReadFile("./remember.json")
	if err != nil {
		fmt.Println("File does not exist!!")
		return
	}

	details := map[string]interface{}{}
	json.Unmarshal(userFile, &details)
	fmt.Println(details)
}

File: python/chapter10/remember_me.py
import json
from pathlib import Path


def save_user_info():
    name = input("Enter name: ")

    try:
        age = int(input("Enter age: "))
    except ValueError:
        print("Please enter a valid age!!")
        return

    contact = input("Enter contact: ")

    with Path("./remember.json") as file:
        details = json.dumps({"name": name, "age": age, "contact": contact})
        file.write_text(details + "\n")


def display_user_info():
    try:
        file = Path("./remember.json")
        contents = json.loads(file.read_text())
        print(contents)
    except FileNotFoundError:
        print("File does not exist!!")
        return


display_user_info()
save_user_info()
